test/local/commands/actor/calculate-memory.test.ts

```diff
@@ -0,0 +1,93 @@
+import { mkdirSync, writeFileSync } from 'node:fs';
+import { mkdir } from 'node:fs/promises';
+import { join } from 'node:path';
+
+import { ActorCalculateMemoryCommand } from '../../../../src/commands/actor/calculate-memory.js';
+import { testRunCommand } from '../../../../src/lib/command-framework/apify-command.js';
+import { EMPTY_LOCAL_CONFIG, LOCAL_CONFIG_PATH } from '../../../../src/lib/consts.js';
+import { useConsoleSpy } from '../../../__setup__/hooks/useConsoleSpy.js';
+import { useTempPath } from '../../../__setup__/hooks/useTempPath.js';
+
+const { beforeAllCalls, afterAllCalls, joinPath, toggleCwdBetweenFullAndParentPath } = useTempPath('calculate-memory', {
+	create: true,
+	remove: true,
+	cwd: true,
+	cwdParent: false,
+});
+
+const { lastErrorMessage } = useConsoleSpy();
+
+const createActorJson = async (overrides: Record<string, unknown> = {}) => {
+	const actorJson = { ...EMPTY_LOCAL_CONFIG, ...overrides };
+
+	await mkdir(joinPath('.actor'), { recursive: true });
+	writeFileSync(joinPath(LOCAL_CONFIG_PATH), JSON.stringify(actorJson, null, '\t'), { flag: 'w' });
+};
+
+describe('apify actor calculate-memory', () => {
+	beforeEach(async () => {
+		await beforeAllCalls();
+		toggleCwdBetweenFullAndParentPath();
+	});
+
+	afterEach(async () => {
+		await afterAllCalls();
+	});
+
+	it('should fail when default memory is not provided and actor.json is missing', async () => {
+		await testRunCommand(ActorCalculateMemoryCommand, {});
+
+		expect(lastErrorMessage()).toMatch(/actor.json not found at/);
+	});
+
+	it('should fail when default memory is not provided in flags or actor.json', async () => {
+		await createActorJson();
+
+		await testRunCommand(ActorCalculateMemoryCommand, {});
+
+		expect(lastErrorMessage()).toMatch(/No memory-calculation expression found./);
+	});
+
+	it('should calculate memory using defaultMemoryMbytes flag', async () => {
+		await testRunCommand(ActorCalculateMemoryCommand, {
+			flags_input: 'INPUT.json',
+			flags_defaultMemoryMbytes: '512',
```

Comment:
It's confusing that when providing the expression via a flag, you use just a constant value, but when providing from `actor.json`, you use an actual expression. That creates the impression that it works differently but as far as I can tell, it doesn't?

I'd just use the same thing in both (actually all, dtto below) cases.

What you can do is something like:

```ts
const START_URLS_LENGTH_BASED_MEMORY_EXPRESSION = 'get(input, 'startUrls.length') * 1024';
```

And then reuse it across all cases. That will make it super clear that the expression itself doesn't really matter and remains constant across cases.